feat(functions): expand xor function and add or function (#2765)
stoplightio/spectral: docs/reference/functions.md

      reusableObjectsLocation: "#/definitions"
```

## or

> Communicate that one or more of these properties is required to be defined. FunctionOptions must contain any non-zero number of properties, **or** will require that _at least_ one of them is defined. (For only one property specified, this is the same as the `defined` rule for that property.)

Reviewer: ```suggestion
Communicate that one or more of these properties is required to be defined. `functionOptions` must contain at least two properties. **or** will require that _at least_ one of them is defined.
```

<!-- title: functionOptions -->

| name       | description             | type       | required? |
| ---------- | ----------------------- | ---------- | --------- |
| properties | the properties to check | `string[]` | yes       |

<!-- title: example -->

```yaml
schemas-descriptive-text-exists:
  description: Defined schemas must have one or more of `title`, `summary` and/or `description` fields.
  given: "$.components.schemas.*"
  then:
    function: or
    functionOptions:
      properties:
        - title
        - summary
        - description
```

## xor

> Communicate that one of these properties is required, and no more than one is allowed to be defined. FunctionOptions must contain any non-zero number of properties, **xor** will require that _exactly_ one of them is defined. (For only one property specified, this is the same as the `defined` rule for that property.)

Reviewer: ```suggestion
Communicate that one of these properties is required, and no more than one is allowed to be defined. `functionOptions` must contain at least two properties and **xor** will require that _exactly_ one of them is defined.
```
